### PR TITLE
drivers: sensor: adxl345: Updated ADXL345 driver with RTIO stream and Trigger functionality

### PR DESCRIPTION
**Pull request by dimitrije-lilic**
Updated ADXL345 driver with RTIO stream functionality. Added Trigger interrupt functionality. RTIO stream is using FIFO threshold. Together with RTIO stream, RTIO async read is also implemented.

Signed-off-by: Dimitrije Lilic dimitrije.lilic@orioninc.com

### CODE REVIEW

**Comment by pblxptr** [comment]
Why only just one sensor trigger type?

**Comment by dimitrije-lilic** [comment]
> 1667479

Hi @pblxptr,

Main focus here is to add RTIO streaming functionality to the driver, the trigger part was added only for watermark in order to implement and test the previous. I can't answer what are plans for other triggers.

File: drivers/sensor/adi/adxl345/adxl345.c [hunk 1]
```diff
@@ -54,60 +54,239 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
 		.count = 2,
 	};
 
+	int ret = 2;
```
Comment (MaureenHelm) [inline drivers/sensor/adi/adxl345/adxl345.c:57]: Why assign to 2?

File: drivers/sensor/adi/adxl345/adxl345.c [hunk 2]
```diff
@@ -54,60 +54,239 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
...
+int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
 
 {
 	return adxl345_reg_read(dev, addr, buf, 1);
 }
 
+int adxl345_reg_write_mask(const struct device *dev,
+			       uint8_t reg_addr,
+			       uint32_t mask,
+			       uint8_t data)
```
Comment (MaureenHelm) [inline drivers/sensor/adi/adxl345/adxl345.c:105]: Why is the mask wider than the data?

**Review by MaureenHelm — COMMENTED** [review]
The clang-format warnings are noisy, but there are some real checkpatch errors that need to be addressed.

File: drivers/sensor/adi/adxl345/adxl345.c (hunk 1 above)
Comment (dimitrije-lilic) [reply drivers/sensor/adi/adxl345/adxl345.c]: That was leftover from debugging, and is removed now

File: drivers/sensor/adi/adxl345/adxl345.c (hunk 2 above)
Comment (dimitrije-lilic) [reply drivers/sensor/adi/adxl345/adxl345.c]: It's now set to uint8_t mask, It was my oversight on my part.

File: drivers/sensor/adi/adxl345/adxl345_decoder.c [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	[ADXL345_ODR_400HZ] = UINT32_C(1000000000) / 400,
+};
+
+static inline void adxl345_accel_convert_q31(q31_t *out, uint16_t sample)
+{
+	if (sample & BIT(9)) {
+		sample |= ADXL345_COMPLEMENT;
+	}
+	int32_t micro_ms2 = ((sample * SENSOR_G) / 32);
+	*out = CLAMP((((int64_t)micro_ms2) + (micro_ms2 % 1000000)), INT32_MIN, INT32_MAX);
```
Comment (ubieda) [inline drivers/sensor/adi/adxl345/adxl345_decoder.c:28]: Similar to https://github.com/zephyrproject-rtos/zephyr/pull/78941#discussion_r1811265047

**Review by ubieda — APPROVED** [review]
+1 to get this in. We can iterate on decoding separatedly (https://github.com/zephyrproject-rtos/zephyr/pull/78941#discussion_r1811277602)

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Comment by teburd** [comment]
> +1 to get this in. We can iterate on decoding separatedly ([#78941 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/78941#discussion_r1811277602))

Sounds good, I trust there will be some follow ups

**Pull request merged**